pyro/contrib/gp/models/sgpr.py

```diff
@@ -139,6 +139,7 @@ def model(self):
                 D = D + Kffdiag - Qffdiag
             else:  # approx = "VFE"
                 trace_term = (Kffdiag - Qffdiag).sum() / self.noise
+                trace_term = trace_term.clamp(min=0)
```

Comment:
This is to guarantee `trace_term` nonnegative. If it is negative, `pyro.sample("trace_term", dist.Bernoulli(probs=torch.exp(-trace_term / 2.))` will throw error because `probs > 1`.